internal/release/cross/release_list.go

```diff
@@ -163,9 +163,10 @@ func (r *ReleaseList) GetReleasesForPromotion(sourceEnv, targetEnv *v1alpha1.Env
 	return subset, nil
 }
 
-// HasNonPromotableReleases checks if the list contains releases that cannot be promoted based
+// GetInvalidReleaseVersions returns a list of names of releases that cannot be promoted based
 // on the version format allowed at the target environment.
-func (r *ReleaseList) HasNonPromotableReleases(sourceEnv, targetEnv *v1alpha1.Environment) bool {
+func (r *ReleaseList) GetInvalidReleaseVersions(sourceEnv, targetEnv *v1alpha1.Environment) []string {
```

Comment:
Haha, I know it's David who proposed that name, but I disagree, because those are not invalid, they are just pre-release. I would opt for something more aligned with the real semantic/intent of the function, very similar to what you had initially (which was good, though David suggested to get the specific list) so something like `GetNonPromotableReleases`.